### ISSUE TITLE
NTv3 species conditioning does not support <mask> tokens ?

### ISSUE THREAD

**Issue by eyes-robson** [instadeepai/nucleotide-transformer#136]
Per the bioRxiv preprint, 

> For _Solanum lycopersicum_, which was not included in the post-training dataset, we used the <mask> conditioning token to enable generalization to unseen species.

However, this does not seem to be presently supported by the NTv3 codebase. Was something else done for predicting unseen species? Or can this be functionality be re-enabled?

Specifically, **two** assertion errors are encountered, the **first** when attempting to call [model.encode_species](https://github.com/instadeepai/nucleotide-transformer/blob/2dc37b86e16a6970fbc731751f7719d9f676f7f9/nucleotide_transformer_v3/model.py#L1068), which returns an assertion error `Unknown species '{s}'. Supported species: {self.supported_species}`. This is probably because [`self.supported_species`](https://github.com/instadeepai/nucleotide-transformer/blob/2dc37b86e16a6970fbc731751f7719d9f676f7f9/nucleotide_transformer_v3/model.py#L1066) specifically filters out any token beginning with "<". 

The **second** assertion error, `invalid species_ids: species_tokens must be greater than or equal to`, occurs when attempting to bypass this encoder by manually encoding the `<mask>` token, as there is a comparison against [`self.config.num_species_special_tokens`](https://github.com/instadeepai/nucleotide-transformer/blob/2dc37b86e16a6970fbc731751f7719d9f676f7f9/nucleotide_transformer_v3/model.py#L1032), which explicitly prohibits the model from using a <mask> token.

Is there an alternative for predicting unseen species?

**Comment by bernardo-de-almeida** [comment]
Hi @eyes-robson , thanks for your comment. You are right, we did that to make it simpler for the users of the model and to prevent mis-use by mistake.

For fine-tuning NTv3 on new species, please see our notebook [03_fine_tuning_posttrained_model_biwig.ipynb](https://huggingface.co/spaces/InstaDeepAI/ntv3/blob/main/notebooks_tutorials/03_fine_tuning_posttrained_model_biwig.ipynb) where we show how to handle new species and test it on the tomato benchmark.

Let me know if you have any further question
